include/mgp.hpp

```diff
@@ -2436,12 +2389,24 @@ inline mgp::Id Relationship::Id() const { return Id::FromInt(mgp::edge_get_id(pt
 
 inline std::string_view Relationship::Type() const { return mgp::edge_get_type(ptr_).name; }
 
-inline class Properties Relationship::Properties() const {
-  return mgp::Properties(mgp::edge_iter_properties(ptr_, memory));
+inline std::map<std::string, Value> Relationship::Properties() const {
+  mgp_properties_iterator *properties_iterator = mgp::edge_iter_properties(ptr_, memory);
+  std::map<std::string, Value> property_map_;
+  for (mgp_property *property = mgp::properties_iterator_get(properties_iterator); property;
+       property = mgp::properties_iterator_next(properties_iterator)) {
+    Value value = Value(property->value);
+    property_map_.emplace(property->name, value);
+  }
+  mgp::properties_iterator_destroy(properties_iterator);
+  return property_map_;
+}
+
+inline void Relationship::SetProperty(std::string property, Value value) const {
+  mgp::edge_set_property(ptr_, property.data(), value.ptr());
 }
 
-inline Value Relationship::operator[](const std::string_view property_name) const {
-  return Properties()[property_name];
+inline Value Relationship::GetProperty(const std::string &property) const {
+  return Value(mgp::edge_get_property(ptr_, property.data(), memory));
```

Comment:
I think the same approach can be used in `Node::GetProperty` also.

Reply:
Problem is with function get_property which makes already copy of mgp_value. If we make another copy, then there will be 2 copies and one won't get deleted because we didn't use mgp_value_destory and one of them. I forgot to apply that here, so I am adding it now.